src/main/java/org/assertj/core/api/AbstractThrowableAssert.java

```diff
@@ -119,6 +120,32 @@ public SELF hasCause(Throwable cause) {
     return myself;
   }
 
+  /**
+   * Verifies that the actual {@code Throwable} has a cause the same as the given one, i.e. using == comparison
+   * <p>
+   * Example:
+   * <pre><code class='java'>
+   * Throwable invalidArgException = new IllegalArgumentException("invalid arg");
+   * Throwable throwable = new Throwable(invalidArgException);
+   *
+   * // This assertion succeeds:
+   * assertThat(throwable).hasCauseSameAs(invalidArgException);
+   *
+   * // These assertions fail:
+   * assertThat(throwable).hasCauseSameAs(new IllegalArgumentException("invalid arg"));
+   * assertThat(throwable).hasCauseSameAs(new NullPointerException());
+   * assertThat(throwable).hasCauseSameAs(null); // prefer hasNoCause()</code></pre>
+   *
+   * @param cause the expected cause
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code Throwable} is {@code null}.
+   * @throws AssertionError if the actual {@code Throwable} has not the given cause.
+   */
+  public SELF hasCauseSameAs(Throwable cause) {
```

Comment:
reword to `hasSameCauseAs` and update the cod examples accordingly

Reply:
Yea I struggled with the naming a bit, but `hasCauseReference` sounds good to me. I'll update accordingly.